bangc-ops/kernels/prior_box/prior_box.cpp

```diff
@@ -0,0 +1,224 @@
+  if (max_sizes_desc->total_element_num != 0) {
+    num_priors += max_sizes_desc->dims[0];
+  }
+  return num_priors;
+}
+
+mluOpStatus_t mluOpPriorBoxParamCheck(
+    mluOpHandle_t handle, const mluOpTensorDescriptor_t min_sizes_desc,
+    const void *min_sizes, const mluOpTensorDescriptor_t aspect_ratios_desc,
+    const void *aspect_ratios, const mluOpTensorDescriptor_t variances_desc,
+    const void *variances, const mluOpTensorDescriptor_t max_sizes_desc,
+    const void *max_sizes, const int height, const int width,
+    const int im_height, const int im_width, const float step_h,
+    const float step_w, const float offset, bool clip,
+    bool min_max_aspect_ratios_order, const mluOpTensorDescriptor_t output_desc,
+    void *output, const mluOpTensorDescriptor_t var_desc, void *var) {
+  // check input
+  PARAM_CHECK(api, handle != nullptr);
+  PARAM_CHECK(api, min_sizes_desc != nullptr);
+  PARAM_CHECK(api, aspect_ratios_desc != nullptr);
+  PARAM_CHECK(api, variances_desc != nullptr);
+  PARAM_CHECK(api, max_sizes_desc != nullptr);
+  // check dim
+  PARAM_CHECK(api, min_sizes_desc->dim == 1);
+  PARAM_CHECK(api, aspect_ratios_desc->dim == 1);
+  PARAM_CHECK(api, variances_desc->dim == 1);
+  PARAM_CHECK(api, max_sizes_desc->dim == 1);
+  PARAM_CHECK(api, output_desc->dim == 4);
+  PARAM_CHECK(api, var_desc->dim == 4);
+  // check shape
+  PARAM_CHECK(api, variances_desc->dims[0] == 4);
+  PARAM_CHECK_GT(api, height, 0);
+  PARAM_CHECK_GT(api, width, 0);
+  PARAM_CHECK_GT(api, im_height, 0);
+  PARAM_CHECK_GT(api, im_width, 0);
+  // check data type
+  PARAM_CHECK(api, min_sizes_desc->dtype == MLUOP_DTYPE_FLOAT);
+  PARAM_CHECK(api, aspect_ratios_desc->dtype == MLUOP_DTYPE_FLOAT);
+  PARAM_CHECK(api, variances_desc->dtype == MLUOP_DTYPE_FLOAT);
+  PARAM_CHECK(api, max_sizes_desc->dtype == MLUOP_DTYPE_FLOAT);
+  PARAM_CHECK(api, output_desc->dtype == MLUOP_DTYPE_FLOAT);
+  PARAM_CHECK(api, var_desc->dtype == MLUOP_DTYPE_FLOAT);
+  // check scalar param
+  PARAM_CHECK_GT(api, step_h, 0);
+  PARAM_CHECK_GT(api, step_w, 0);
+  // check param depand
+  PARAM_CHECK(api, output_desc->dims[0] == height);
+  PARAM_CHECK(api, output_desc->dims[1] == width);
+  for (int i = 0; i < output_desc->dim; i++) {
+    PARAM_CHECK(api, output_desc->dims[i] == var_desc->dims[i]);
+  }
+  if (max_sizes_desc->total_element_num != 0) {
+    PARAM_CHECK(api, max_sizes_desc->dims[0] == min_sizes_desc->dims[0]);
+    PARAM_CHECK(api,
+                max_sizes_desc->dims[0] +
+                        min_sizes_desc->dims[0] * aspect_ratios_desc->dims[0] ==
+                    output_desc->dims[2]);
+  } else {
+    PARAM_CHECK(api, min_sizes_desc->dims[0] * aspect_ratios_desc->dims[0] ==
+                         output_desc->dims[2]);
+  }
+  const int num_priors =
+      getNumPriors(min_sizes_desc, aspect_ratios_desc, max_sizes_desc);
+  // check num_priors limit
+  const int max_support_num_priors = getMaxSupportNumPriors(handle);
```

Comment:
建议删除getMaxSupportNumPriors
```suggestion
  const int max_support_num_priors = 
      handle->arch < 300 ? MLU200SERIERS_MAX_SUPPORT : MLU300SERIERS_MAX_SUPPORT;;
```